components/net/fetch/methods.rs

```diff
@@ -237,62 +237,148 @@ pub fn main_fetch(request: Rc<Request>,
 
     // Step 13
     // no need to check if response is a network error, since the type would not be `Default`
-    let response = if response.response_type == ResponseType::Default {
+    if response.response_type == ResponseType::Default {
         let response_type = match request.response_tainting.get() {
             ResponseTainting::Basic => ResponseType::Basic,
             ResponseTainting::CorsTainting => ResponseType::Cors,
             ResponseTainting::Opaque => ResponseType::Opaque,
         };
-        response.to_filtered(response_type)
+        response = response.to_filtered(response_type)
+    }
+
+    // Step 14
+    let mut internal_response = if let Some(error) = response.get_network_error() {
+        Response::network_error(error.clone())
     } else {
-        response
+        response.actual_response().clone()
```

Comment:
This is **NOT** an improvement. We purposely avoided cloning because it's an expensive operation to do so for a Response.